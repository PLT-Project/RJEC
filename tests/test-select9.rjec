/* NOTE: might be race conditions causing variations of the test output */

func foo (x []chan int){
    for {
        select {
        case i := <- x[0]:
            printi(i);
        case j := <- x[1]:
            printi(j);
        case x[2] <- 1:
            prints("sent 1 through c2");
        case x[3] <- 2:
        }
    }
}

func main () {
    x := []chan int {
        make(chan int),
        make(chan int, 2),
        make(chan int, 2),
        make(chan int)
    };

    yeet foo(x);
    
    x[0] <- 1;
    x[0] <- 2;
    x[1] <- 10;
    printi(<- x[3]);
    printi(<- x[3]);
}