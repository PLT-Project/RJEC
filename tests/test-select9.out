sent 1 through c2
sent 1 through c2
1
2
10
2
2